File: application/src/main/java/org/sheedon/common/handler/ILoadingDialogHandler.java
package org.sheedon.common.handler;

import android.content.Context;

import org.sheedon.common.widget.LoadingDialog;

/**
 * 加载弹窗处理者
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/10/19 6:42 下午
 */
public interface ILoadingDialogHandler {

    /**
     * 显示描述弹窗
     *
     * @param message 描述
     */
    void showLoading(String message);

    /**
     * 隐藏弹窗
     */
    void hideLoading();


    class LoadingDialogFactory {

        public ILoadingDialogHandler createLoadingDialog(Context context) {
            return new LoadingDialog(context);
        }

    }

    /**
     * 加载框执行类
     */
    class LoadingDialogHandler {
        private static final LoadingDialogHandler INSTANCE = new LoadingDialogHandler();
        // 加载框创建工厂
        private LoadingDialogFactory factory;

        private LoadingDialogHandler() {

        }

        public static LoadingDialogHandler getInstance() {
            return INSTANCE;
        }

        /**
         * 附加替换加载弹窗工厂
         *
         * @param factory 弹窗工厂
         */
        public void attachLoadingDialogFactory(LoadingDialogFactory factory) {
            this.factory = factory;
        }

        /**
         * 获取弹窗工厂类，用于调用者通过该类创建弹窗
         */
        public LoadingDialogFactory getFactory() {
            if (factory == null) {
                factory = new LoadingDialogFactory();
            }
            return factory;
        }
    }

}

File: application/src/main/java/org/sheedon/common/widget/loading/ILoadingDialogHandler.kt
package org.sheedon.common.widget.loading

import android.content.Context

/**
 * 加载弹窗处理者
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/5 2:12 下午
 */
interface ILoadingDialogHandler {

    /**
     * 显示描述弹窗
     *
     * @param message 描述
     */
    fun showLoading(message: String? = null)

    /**
     * 隐藏弹窗
     */
    fun hideLoading()

    open class LoadingDialogFactory {

        protected open fun createLoadingDialog(content: Context): ILoadingDialogHandler {
            return LoadingDialog(content)
        }
    }

    /**
     * 加载框执行类
     */
    class LoadingDialogHandler private constructor() {

        // 加载框创建工厂
        var factory: LoadingDialogFactory? = null
            get() {
                // 获取弹窗工厂类，用于调用者通过该类创建弹窗
                if (field == null) {
                    field = LoadingDialogFactory()
                }
                return field
            }
            private set

        companion object {
            private val INSTANCE = LoadingDialogHandler()

            @JvmStatic
            fun getInstance() = INSTANCE
        }

        /**
         * 附加替换加载弹窗工厂
         *
         * @param factory 弹窗工厂
         */
        fun attachLoadingDialogFactory(factory: LoadingDialogFactory) {
            this.factory = factory
        }
    }

}